wkcuber/export_wkw_as_nifti.py

```diff
@@ -0,0 +1,132 @@
+import logging
+from argparse import ArgumentParser, Namespace
+from pathlib import Path
+from typing import Dict, List, Optional
+
+import nibabel as nib
+import numpy as np
+
+from wkcuber.api.bounding_box import BoundingBox
+from wkcuber.api.Dataset import WKDataset
+from wkcuber.mag import Mag
+from wkcuber.metadata import read_metadata_for_layer
+from wkcuber.utils import add_distribution_flags, add_verbose_flag, parse_bounding_box
+
+
+def create_parser() -> ArgumentParser:
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "--source_path", "-s", help="Directory containing the wkw file.", required=True
+    )
+
+    parser.add_argument(
+        "--destination_path",
+        "-d",
+        help="Output directory for the generated nifti files.",
+        required=True,
+    )
+
+    parser.add_argument("--name", "-n", help="Name of the nifti", default="")
+
+    parser.add_argument(
+        "--bbox",
+        help="The BoundingBox of which the nifti file should be generated."
+        "The input format is x,y,z,width,height,depth."
+        "(By default, data for the full bounding box of the dataset is generated)",
+        default=None,
+        type=parse_bounding_box,
+    )
+
+    parser.add_argument(
+        "--mag", "-m", help="The magnification that should be read", default=1
+    )
+
+    parser.add_argument(
+        "--downsample", help="Downsample each nifti image", default=1, type=int
+    )
+
+    # add_batch_size_flag(parser)
+
+    add_verbose_flag(parser)
+    add_distribution_flags(parser)
+
+    return parser
+
+
+def export_layer_to_nifti(
+    wkw_file_path: Path,
+    bbox: Dict,
+    mag: Mag,
+    layer_name: str,
+    destination_path: Path,
+    name: str,
+) -> None:
+    wk_ds = WKDataset(wkw_file_path)
+    layer = wk_ds.get_layer(layer_name)
+    mag_layer = layer.get_mag(mag)
+    data = mag_layer.read(bbox["topleft"], bbox["size"])
+    data = data.transpose(1, 2, 3, 0)
+    # Problem: different dtypes (uint8 and uint32), implicit cast
+    logging.info(f"Shape with layer {data.shape}")
+
+    data = np.array(data)
+    img = nib.Nifti1Image(data, np.eye(4))
+
+    destination_file = str(destination_path.joinpath(name + ".nii"))
+
+    logging.info(f"Writing to {destination_file} with shape {data.shape}")
+
+    nib.save(img, destination_file)
+
+
+def export_nifti(
+    wkw_file_path: Path,
+    bbox: Optional[BoundingBox],
+    mag: Mag,
+    destination_path: Path,
+    name: str,
+) -> None:
+    wk_ds = WKDataset(wkw_file_path)
+    layers = wk_ds.layers
+
+    for layer_name in layers:
+        if bbox is None:
+            _, _, bbox_dim, origin = read_metadata_for_layer(wkw_file_path, layer_name)
```

Comment:
I think these properties should be available via `wk_ds.layers`. @rschwanhold might be able to give more precise hints :)

Reply:
You can use `wk_ds.properties.get_bounding_box_of_layer(...)`

Reply:
FYI: The Dataset API was changed a lot in the recent past. You can now get the bbox of a layer using `Layer.bounding_box`

Reply:
@rschwanhold is this still up-to-date? I can't find this function in the `properties` source code. If it is not, what is the right way to do it?